samples/drivers/i2s/echo/src/main.c

```diff
@@ -0,0 +1,304 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include "codec.h"
+#include <sys/printk.h>
+#include <drivers/i2s.h>
+#include <drivers/gpio.h>
+#include <string.h>
+
+#if DT_NODE_EXISTS(DT_NODELABEL(i2s_rxtx))
+#define I2S_RX_NODE  DT_NODELABEL(i2s_rxtx)
+#define I2S_TX_NODE  I2S_RX_NODE
+#else
+#define I2S_RX_NODE  DT_NODELABEL(i2s_rx)
+#define I2S_TX_NODE  DT_NODELABEL(i2s_tx)
+#endif
+
+#define SAMPLE_FREQUENCY    44100
+#define SAMPLE_BIT_WIDTH    16
+#define BYTES_PER_SAMPLE    sizeof(int16_t)
+#define NUMBER_OF_CHANNELS  2
+/* Such block length provides an echo with the delay of 100 ms. */
+#define SAMPLES_PER_BLOCK   ((SAMPLE_FREQUENCY / 10) * NUMBER_OF_CHANNELS)
+#define INITIAL_BLOCKS      3
+#define TIMEOUT             1000
+
+#define SW0_NODE        DT_ALIAS(sw0)
+#if DT_NODE_HAS_STATUS(SW0_NODE, okay)
+#define SW0_GPIO_CTLR   DT_GPIO_CTLR(SW0_NODE, gpios)
```

Comment:
Please use a `struct gpio_dt_spec`. I have an example in ` samples: button: devicetree cleanups ` in https://github.com/zephyrproject-rtos/zephyr/pull/34589 that relies on a new macro for potentially missing nodes, but this could be done even with a plain GPIO_DT_SPEC_GET and some extra ifdefs, and it would still be simpler.

Reply:
Applied.